test-data/unit/check-attr.test

```diff
@@ -1428,51 +1428,76 @@ class B(A):
 reveal_type(B)  # N: Revealed type is "def (foo: builtins.int) -> __main__.B"
 [builtins fixtures/bool.pyi]
 
-[case testAttrsClassHasAttributeWithAttributes]
+[case testAttrsClassHasMagicAttribute]
 import attr
 
 @attr.s
 class A:
     b: int = attr.ib()
     c: str = attr.ib()
 
-reveal_type(A.__attrs_attrs__)  # N: Revealed type is "Tuple[attr.Attribute[builtins.int], attr.Attribute[builtins.str], fallback=__main__.A._AttrsAttributes]"
+reveal_type(A.__attrs_attrs__)  # N: Revealed type is "Tuple[attr.Attribute[builtins.int], attr.Attribute[builtins.str], fallback=__main__.A.____main___A_AttrsAttributes__]"
```

Comment:
This part `____main___A_AttrsAttributes__` looks very scary.

I hope it won't be user visible :)

Reply:
I know it sucks. The reason for it is: when loading the type information from cache, Mypy seems to need to be able to find the class using a fully qualified path. So I'm trying to avoid collisions by stuffing more information into the name. Open to suggestions ;)